can-map.js

```diff
@@ -601,6 +604,30 @@ var Map = Construct.extend(
 
 		},
 
+		// ### compute
+		// Creates a compute that represents a value on this map. If the property is a function
+		// on the prototype, a "function" compute wil be created.
+		// Otherwise, a compute will be created that reads the observable attributes
+		compute: function (prop) {
+			if (isFunction(this.constructor.prototype[prop])) {
+				return canCompute(this[prop], this);
+			} else {
+				var reads = ObserveReader.reads(prop);
+				var last = reads.length - 1;
+
+				return canCompute(function (newVal) {
+					if (arguments.length) {
+						ObserveReader.read(this, reads.slice(0, last))
+							.value.attr(reads[last].key, newVal);
+					} else {
+						return ObserveReader.read(this, reads, {
```

Comment:
And I think there's an ObserveReader.get where you can just pass a string instead of `reads`.